tuf/api/metadata.py

```diff
@@ -1213,8 +1213,12 @@ def __init__(
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ):
         self.keys = keys
-        if [role for role in set(roles) if role in TOP_LEVEL_ROLE_NAMES]:
-            raise ValueError("Delegated roles cannot use top-level role names")
+
+        for role in set(roles):
+            if not role or role in TOP_LEVEL_ROLE_NAMES:
+                raise ValueError(
+                    "Delegated roles cannot be empty or use top-level role names"
+                )
```

Comment:
You still can write the check as a one-line like this:
```suggestion
        if any(not role or role in TOP_LEVEL_ROLE_NAMES for role in roles):
            raise ValueError(
                "Delegated roles cannot be empty or use top-level role names"
            )
```
and I personally prefer it.
@sechkova what do you think?

Reply:
Yes, I had a small talk with @jku, and we came out about making it more readable. I'm open about any option 🙂

Reply:
Thanks for adding the check, I agree that separate lines are more readable.

Reply:
set() does not seem necessary?